src/service/AttachmentService.js

```
@@ -1,29 +1,42 @@
 import axios from 'axios';
+import { buf } from 'crc-32/crc32c';
 
-const baseUrl = 'http://localhost:3001/attachments';
+const baseUrl = 'http://localhost:3001/attachment/';
 
-function upload(attachment, setResponseStatus) {
-  const formData = new FormData();
-  if (attachment) {
-    formData.append('attachment', attachment);
-  }
+function upload(attachment, setResponseStatus, setDownloadUri) {
+  const reader = new FileReader();
 
-  const url = `${baseUrl}/`;
-  const headers = {
-    headers: {
-      'content-type': 'multipart/form-data',
-    },
+  reader.onload = (e) => {
+    const metadata = {
+      name: attachment.name,
+      size: attachment.size,
+      type: attachment.type,
+      crc: buf(new Uint8Array(e.target.result)),
+    };
+
+    const headers = { 'Content-Type': 'application/json', 'Access-Control-Allow-Origin': '*', 'Access-Control-Allow-Headers': '*', 'Access-Control-Allow-Credentials': 'true' };
+
+    axios
+      .post(`${baseUrl}`, metadata, { headers: headers })
+      .then((response) => {
+        axios
+          .put(response.headers.location, e.target.result, { headers: headers })
+          .then(() => {
+            setResponseStatus('success');
+            setDownloadUri(response.headers.location);
+          })
+          .catch((err) => setResponseStatus('error in binary upload', err));
```

This is too much implementation detail for the user - as far as they are concerned the upload did not work.  You should also let the err propagate to the caller so they can react appropriately there - if you swallow the error then any handlers in the promise chain will invoked irrespective of error status.

I would simplify this to a single catch at the end of the change with a user friendly message and a rethrow of the err.

---

Have pushed a fix. will probably move to using a toastify notification that is updated by the response status state so will push that soon